TradeItIosTicketSDK2/TradeItPortfolioPosition.swift

```diff
@@ -22,29 +22,36 @@ class TradeItPortfolioPosition : NSObject {
     }
     
     func refreshQuote(onFinished onFinished: () -> Void) {
-        var tradeItQuoteRequest: TradeItQuotesRequest!
+        var tradeItQuoteRequest: TradeItQuotesRequest?
         var symbol = ""
-        if let position = self.position {
-            symbol = position.symbol
+
+        if let position = self.position, let equitySymbol = position.symbol {
+            symbol = equitySymbol
             tradeItQuoteRequest = TradeItQuotesRequest(symbol: symbol)
         }
-        if let fxPosition = self.fxPosition {
-                symbol = fxPosition.symbol
-                tradeItQuoteRequest = TradeItQuotesRequest(fxSymbol: symbol, andBroker: self.linkedBrokerAccount.brokerName)
+
+        if let fxPosition = self.fxPosition, let fxSymbol = fxPosition.symbol {
+            symbol = fxSymbol
+            tradeItQuoteRequest = TradeItQuotesRequest(fxSymbol: symbol, andBroker: self.linkedBrokerAccount.brokerName)
+        }
+
+        guard (tradeItQuoteRequest != nil) else {
+            onFinished()
+            return
         }
-        var quote = TradeItQuote()
+
         self.tradeItMarketDataService.getQuoteData(tradeItQuoteRequest, withCompletionBlock: { (tradeItResult: TradeItResult!) -> Void in
             if let tradeItQuoteResult = tradeItResult as? TradeItQuotesResult {
-                let results = tradeItQuoteResult.quotes.filter { return $0.symbol == symbol}
-                if results.count > 0 {
-                    quote = results[0] as! TradeItQuote
-                    self.quote = quote
+                let results = tradeItQuoteResult.quotes?.filter { return $0.symbol == symbol }
+
+                if let results = results as? [TradeItQuote] where results.count > 0 {
+                    self.quote = results[0]
```

Comment:
You can also use `self.quote = results.first` which won't explode if there is no element - it will just return `nil`.